### PR TITLE
Refactor BYON images table

### LINKED ISSUE #1432: Update all text items and field labels on Notebook images table
### Goal

Make sure the text items and fields the same as the mock up listed as below

### Dependency issue

Tracker: https://github.com/opendatahub-io/odh-dashboard/issues/1254

### Itemized goals

- Change the nav item title
- Change the page description
- Use the generic table component
- Add a helper tooltip for the `Enable` field on the custom image table
- Change the expandable section UI to align with the design

### Aspects continued elsewhere

Update all text items and field labels on Notebook images modals (Import, Edit, Delete) - ticket TBD

### Mocks

image

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Closes #1432 

## Description

1. Refactor the whole table, using the generic table component
2. use `useFetchState` hook to get BYON images
3. Refactor the usage of the modals. (as for the refactor of the contents in the modals, it will be solved in #1433)

Screenshot 2023-07-12 at 10 47 57 AM

## How Has This Been Tested?

1. Go to the settings -> Notebook image settings
2. Create a notebook image, and try to play around with it, like editing, deleting, toggle the enablement... make sure nothing is broken

## Test Impact

TBD

## Request review criteria:

Self checklist (all need to be checked):
- [x] The developer has manually tested the changes and verified that the changes work
- [x] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [x] Included any necessary screenshots or gifs if it was a UI change.
- [x] Included tags to the UX team if it was a UI/UX change.

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

**Comment by vconzola** [comment]
@DaoDaoNoCode Does this PR address including out-of-box images in the table - it doesn't look like it does? Will that be handled in a follow-up PR? Also, I don't see this spelled out in the issue requirements, but since we're not including click/drag yet, would it be a simple change (" one liner") to add sorting to the Enable column? That would allow users to see all of their enabled images at the top.

File: frontend/src/pages/BYONImages/tableData.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  },
+  {
+    field: 'description',
+    label: 'Description',
+    sortable: (a, b) => a.description.localeCompare(b.description),
+  },
+  {
+    field: 'enable',
+    label: 'Enable',
+    sortable: false,
```
Comment (andrewballantyne) [inline frontend/src/pages/BYONImages/tableData.tsx:23]: Actually, it should be a one liner @vconzola haha.

File: backend/src/types.ts [hunk 2]
```diff
@@ -466,36 +466,23 @@ export type ODHSegmentKey = {
...
-  BYONImageUpdateRequest;
-
-export type BYONImageCreateRequest = {
+  provider: string;
+  imported_time: string;
+  error: string;
   name: string;
   url: string;
-  description?: string;
-  // FIXME: This shouldn't be a user defined value consumed from the request payload but should be a controlled value from an authentication middleware.
```
Comment (manaswinidas) [inline backend/src/types.ts:479]: Is this taken care of already?

**Review by manaswinidas — COMMENTED** [review]
My only concerns are regarding pagination, how are we handling 20+ images without pagination?

File: backend/src/types.ts (hunk 2 above)
Comment (manaswinidas) [reply backend/src/types.ts:479]: Not sure if we should be removing this comment then. If we are, we should be tracking it somewhere. Are we already tracking this?

Comment (DaoDaoNoCode) [reply backend/src/types.ts:479]: No, i think we should just keep the comment here until we get to refactor the backend in phase 2.

**Review by manaswinidas — COMMENTED** [review]
/lgtm

File: backend/src/types.ts (hunk 2 above)
Comment (manaswinidas) [reply backend/src/types.ts:479]: please keep the comment then, thanks

**Comment by DaoDaoNoCode** [comment]
@vconzola The sorting or `Enable` column will be addressed along with this issue #1388 because we need to make some changes to the Enable toggle.

**Comment by DaoDaoNoCode** [comment]
After the discussion with @vconzola , we are trying to keep the table as it was before - which means we will not add the pagination for now.

**Pull request merged**